### ISSUE TITLE
Train geometry-aware Cd surrogate from voxelized meshes

### ISSUE THREAD

**Issue by MarcosAsh** [MarcosAsh/Lattice_Fluid_Dynamics#149]
Train a 3D CNN that predicts Cd/Cl directly from voxelized mesh geometry + flow parameters, generalizing to arbitrary OBJ shapes.

### Architecture

Input: 3D occupancy grid (1x32x32x32 or 1x64x64x64) + flow parameters (wind_speed, reynolds)

Option A -- 3D CNN encoder + MLP head:
- 3D Conv layers (e.g., 1->16->32->64->128 with stride-2 downsampling)
- Global average pooling to get a fixed-size geometry embedding
- Concatenate with flow parameters
- MLP head: embedding+2 -> 256 -> 128 -> 2 (Cd, Cl)

Option B -- 3D ResNet-style encoder:
- Residual blocks with batch norm for deeper networks
- Same pooling + MLP head structure

### Training data

Generate a diverse dataset by:
1. Using the existing 3 models (car, ahmed25, ahmed35) at multiple wind speeds and Re
2. FFD-deforming each model with random control point displacements (reuse the FFD library from #24)
3. Downloading free OBJ models from online repositories (cars, trucks, simple shapes)
4. Running LBM simulations on Modal for each shape/condition pair
5. Target: 1000-5000 (voxel_grid, wind_speed, reynolds, cd, cl) training samples

### Data format

Binary dataset (.voxbin):
```
Header: magic, version, num_samples, grid_res, num_flow_params, num_outputs
Per sample:
  uint8[grid_res^3]     occupancy grid
  float32[num_flow_params]  [wind_speed, reynolds]
  float32[num_outputs]      [cd, cl]
```

### Training

- PyTorch with 3D convolutions (torch.nn.Conv3d)
- MSE loss on Cd + weighted Cl
- Data augmentation: random Y-axis flips (symmetric shapes), small random rotations
- Train/val/test split: 70/15/15
- Export trained encoder weights for C/GLSL inference

### Success criteria

- Cd MAE < 0.05 on held-out test shapes (shapes not seen during training)
- Generalize across shape categories (not just ahmed body variants)
- Inference < 10ms on GPU for real-time prediction

Depends on #148 (voxelization). Replaces the current surrogate (#22) which only handles 3 fixed model IDs.
